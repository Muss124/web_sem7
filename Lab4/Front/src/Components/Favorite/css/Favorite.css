.FavoriteContainer {
    width          : 99%;
    display        : flex;
    flex-wrap      : wrap;
    flex-direction : row;
    justify-content: space-around;
}