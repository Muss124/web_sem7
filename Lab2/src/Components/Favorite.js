import React from 'react';
import { FavoriteSearch } from "./FavoriteSearch.js"
import { connect } from 'react-redux'
import { FavoriteItem } from "./FavoriteItem.js"

class Favorite extends React.Component {
    render() {
        return (
            <div>
                <FavoriteSearch />
                {
                    this.props.data.map((favCity, i = 0) => {
                        i++;
                        return ([
                            <div class="FavoriteContainer" key={i}>
                                <FavoriteItem value={favCity}  />
                            </div>
                        ])
                    })
                }

            </div>
        )
    }
}

const mapStateToProps = (state) => {
    console.log(state);
    return {
        data: state
    };
}

const mapDispatchToProps = (dispatch) => {
    return {
        setName: () => {
            dispatch({
            });
        }
    };
}
export default connect(mapStateToProps, mapDispatchToProps)(Favorite);