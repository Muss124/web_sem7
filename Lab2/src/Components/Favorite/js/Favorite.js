import React from 'react';
import { FavoriteSearch } from "../../FavoriteSearch/js/FavoriteSearch"
import { connect } from 'react-redux'
import { FavoriteItem } from "../../FavoriteItem/js/FavoriteItem"
import '../css/Favorite.css'

class Favorite extends React.Component {
    render() {
        return (
            <div>
                <FavoriteSearch />
                <div className="FavoriteContainer" >
                    {
                        this.props.data.map((favCity, i = 0) => {
                            i++;
                            return ([

                                <FavoriteItem value={favCity} key={i}/>

                            ])
                        })
                    }
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    console.log(state);
    return {
        data: state
    };
}

const mapDispatchToProps = (dispatch) => {
    return {
        setName: () => {
            dispatch({
            });
        }
    };
}
export default connect(mapStateToProps, mapDispatchToProps)(Favorite);