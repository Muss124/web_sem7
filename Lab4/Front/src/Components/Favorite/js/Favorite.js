import React from 'react';
import { FavoriteSearch } from "../../FavoriteSearch/js/FavoriteSearch"
import { connect } from 'react-redux'
import { FavoriteItem } from "../../FavoriteItem/js/FavoriteItem"
import '../css/Favorite.css'
import FetchFavorite from "../../../Actions/FetchFavorite"
import store from "../../../store"

class Favorite extends React.Component {
    constructor(props) {
        super(props);
        store.dispatch(FetchFavorite());
    }
    render() {
        return (
            <div>
                <FavoriteSearch />
                <div className="FavoriteContainer" >
                    {
                        this.props.data.map((favCity, i = 0) => {
                            i++;
                            return ([

                                <FavoriteItem value={favCity} key={i} />

                            ])
                        })
                    }
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    console.log(state);
    return {
        data: state
    };
}

const mapDispatchToProps = (dispatch) => {
    return {
        setName: () => {
            dispatch({
            });
        }
    };
}
export default connect(mapStateToProps, mapDispatchToProps)(Favorite);