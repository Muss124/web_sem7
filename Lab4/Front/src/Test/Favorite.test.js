import React from 'react';
import Favorite from '../Components/Favorite'
import { shallow } from '../enzyme';
import { shallowToJson } from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

const initialState = {}
const mockStore = configureStore()

describe('Testing render Favorite', function () {
    it("Render from empty storage", () => {
        const storage = [];
        const store = mockStore(initialState);
        const tree = shallow(<Favorite value={storage} store={store}/>);
        expect(shallowToJson(tree)).toMatchSnapshot();
    });
    
    it("Render from with loaded element", () => {
        const storage = [{
            "loading": false,
            "city": "Kiev",
            "data": {
                "coord": { "lon": 30.52, "lat": 50.43 },
                "weather": [{ "id": 802, "main": "Clouds", "description": "scattered clouds", "icon": "03n" }],
                "base": "stations",
                "main": { "temp": 6.29, "feels_like": 2.93, "temp_min": 4, "temp_max": 7.78, "pressure": 1017, "humidity": 87 },
                "visibility": 10000,
                "wind": { "speed": 3, "deg": 230 },
                "clouds": { "all": 30 },
                "dt": 1576604944,
                "sys": { "type": 1, "id": 8903, "country": "UA", "sunrise": 1576561989, "sunset": 1576590872 },
                "timezone": 7200,
                "id": 703448,
                "name": "Kiev",
                "cod": 200
            }
        }];
        const store = mockStore(initialState);
        const tree = shallow(<Favorite value={storage} store={store}/>);
        expect(shallowToJson(tree)).toMatchSnapshot();
    });
    it("Render from empty with loading element", () => {
        const storage = [{
            "loading": true,
            "city": "TEST LOADING CITY",
            "data": {}
        }];
        const store = mockStore(initialState);
        const tree = shallow(<Favorite value={storage} store={store}/>);
        expect(shallowToJson(tree)).toMatchSnapshot();
    });
    it("Render from empty with both loading and loaded element", () => {
        const storage = [
            {
                "loading": false,
                "city": "Kiev",
                "data": {
                    "coord": { "lon": 30.52, "lat": 50.43 },
                    "weather": [{ "id": 802, "main": "Clouds", "description": "scattered clouds", "icon": "03n" }],
                    "base": "stations",
                    "main": { "temp": 6.29, "feels_like": 2.93, "temp_min": 4, "temp_max": 7.78, "pressure": 1017, "humidity": 87 },
                    "visibility": 10000,
                    "wind": { "speed": 3, "deg": 230 },
                    "clouds": { "all": 30 },
                    "dt": 1576604944,
                    "sys": { "type": 1, "id": 8903, "country": "UA", "sunrise": 1576561989, "sunset": 1576590872 },
                    "timezone": 7200,
                    "id": 703448,
                    "name": "Kiev",
                    "cod": 200
                }
            },
            {
                "loading": true,
                "city": "TEST LOADING CITY",
                "data": {}
            }
        ];
        const store = mockStore(initialState);
        const tree = shallow(<Favorite value={storage} store={store}/>);
        expect(shallowToJson(tree)).toMatchSnapshot();
    });
    
});
