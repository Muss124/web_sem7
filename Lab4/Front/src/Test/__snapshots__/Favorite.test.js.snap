// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing render Favorite Render from empty storage 1`] = `
<ContextProvider
  value={null}
>
  <Favorite
    data={Object {}}
    setName={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    value={Array []}
  />
</ContextProvider>
`;

exports[`Testing render Favorite Render from empty with both loading and loaded element 1`] = `
<ContextProvider
  value={null}
>
  <Favorite
    data={Object {}}
    setName={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    value={
      Array [
        Object {
          "city": "Kiev",
          "data": Object {
            "base": "stations",
            "clouds": Object {
              "all": 30,
            },
            "cod": 200,
            "coord": Object {
              "lat": 50.43,
              "lon": 30.52,
            },
            "dt": 1576604944,
            "id": 703448,
            "main": Object {
              "feels_like": 2.93,
              "humidity": 87,
              "pressure": 1017,
              "temp": 6.29,
              "temp_max": 7.78,
              "temp_min": 4,
            },
            "name": "Kiev",
            "sys": Object {
              "country": "UA",
              "id": 8903,
              "sunrise": 1576561989,
              "sunset": 1576590872,
              "type": 1,
            },
            "timezone": 7200,
            "visibility": 10000,
            "weather": Array [
              Object {
                "description": "scattered clouds",
                "icon": "03n",
                "id": 802,
                "main": "Clouds",
              },
            ],
            "wind": Object {
              "deg": 230,
              "speed": 3,
            },
          },
          "loading": false,
        },
        Object {
          "city": "TEST LOADING CITY",
          "data": Object {},
          "loading": true,
        },
      ]
    }
  />
</ContextProvider>
`;

exports[`Testing render Favorite Render from empty with loading element 1`] = `
<ContextProvider
  value={null}
>
  <Favorite
    data={Object {}}
    setName={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    value={
      Array [
        Object {
          "city": "TEST LOADING CITY",
          "data": Object {},
          "loading": true,
        },
      ]
    }
  />
</ContextProvider>
`;

exports[`Testing render Favorite Render from with loaded element 1`] = `
<ContextProvider
  value={null}
>
  <Favorite
    data={Object {}}
    setName={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    value={
      Array [
        Object {
          "city": "Kiev",
          "data": Object {
            "base": "stations",
            "clouds": Object {
              "all": 30,
            },
            "cod": 200,
            "coord": Object {
              "lat": 50.43,
              "lon": 30.52,
            },
            "dt": 1576604944,
            "id": 703448,
            "main": Object {
              "feels_like": 2.93,
              "humidity": 87,
              "pressure": 1017,
              "temp": 6.29,
              "temp_max": 7.78,
              "temp_min": 4,
            },
            "name": "Kiev",
            "sys": Object {
              "country": "UA",
              "id": 8903,
              "sunrise": 1576561989,
              "sunset": 1576590872,
              "type": 1,
            },
            "timezone": 7200,
            "visibility": 10000,
            "weather": Array [
              Object {
                "description": "scattered clouds",
                "icon": "03n",
                "id": 802,
                "main": "Clouds",
              },
            ],
            "wind": Object {
              "deg": 230,
              "speed": 3,
            },
          },
          "loading": false,
        },
      ]
    }
  />
</ContextProvider>
`;
